lib/collection/src/hash_ring.rs

```diff
@@ -121,10 +121,11 @@ impl<T: Hash + Copy> HashRing<T> {
                      but shard {shard} does not exist in the hashring"
                 );
 
-                // TODO(resharding): `return` here?
+                false
             }
-        }
+        };
 
+        // TODO(resharding): Improve old/new hashrings equality check!?
         if old.len() == new.len() {
```

Comment:
Well, we can, but it will be ugly af. Same issue as with `Clone` and `iter`: neither is implemented on `hashring::HashRing`. So to implement `PartialEq` we'd have to clone both hashrings, then call `into_iter` on them and compare elements of the iterators.

Reply:
Let's upstream it!

Reply:
Yeah, my thoughts as well. Seems easier to just send a PR with `#[derive(PartialEq)]`.

Reply:
I've submitted a PR here: https://github.com/jeromefroe/hashring-rs/pull/27

Reply:
I temporarily patched this in 3f7a5e1ff39bfdd547000bf14a6f61e5038cfe28